doc/user/content/transform-data/troubleshooting.md

```diff
@@ -93,11 +95,35 @@ this means that the index was correctly used. If that's not the case, consider:
 * Does the index's indexed expression (key) match up with how you're querying
   the data?
 
-#### Computation-intensive queries
-
-Another thing to be aware of, in Materialize, aggregations like `COUNT(*)` and limit constraints that return more than 25 rows like `LIMIT 50` perform a computation across the full underlying dataset being queried, so can be resource intensive. If you want to execute these types of queries you should either:
-* Create a materialized view or index for these queries if you plan to issue them often
-* Use a large enough cluster to fit the full underlying dataset being queried
+#### Resource-intensive queries
+##### Data-intensive queries
+Queries with aggregations, ordering, or offsets can be data-intensive, because
+they need to read in the full underlying dataset in order to compute the
+result. Some examples are operations like `COUNT()`, `MAX()` and `ORDER BY`.
+If the underlying data is not indexed, then the query execution needs to read
+it in from storage. Ingesting all that data can take time, causing your query
+to run slowly.
+
+##### Computation-intensive queries
+Other queries, like those with `JOIN`s can be computation-intensive.
+Performing those computations on the fly can also take time. During the execution
+of the query, Materialize will temporarily use memory to store the joined
+data and CPU to execute the join. You can read more about the impacts of CPU
+and memory usage in the [cluster CPU](#cluster-cpu) and
+[unhealthy cluster](#unhealthy-cluster) sections.
+
+##### Address
+* If you're looking to explore your data, simple `SELECT ... LIMIT <25 or less>`
+queries on objects will return more quickly. This only applies to queries from a 
+single source, table, or materialized view, with no ordering or offsets.
```

Comment:
Also no filters, as far as I can tell. Perhaps no operations other than maps and projects, though that is just from trying to use each of the three types. Mostly, I worry that `...` is insufficiently spelled out to be clear that it should be very close to `* FROM persist_source` rather than e.g. `DISTINCT col FROM bar` or `* FROM bar WHERE a > 0` or anything like that.

Reply:
> Also no filters, as far as I can tell. Perhaps no operations other than maps and projects, though that is just from trying to use each of the three types.

That's right: map and project are fine, filters are not.

> with no ordering or offsets

If it matters: offset is also okay as long as the offset + limit is less than 25. (It probably doesn't matter; it seems unlikely that a user will use `offset`.)

> Mostly, I worry that ... is insufficiently spelled out

We could phrase this as "simple `SELECT ... FROM <collection> LIMIT <n>` queries will return more quickly, where `<collection>` is a source, table, or materialized view and `<n>` is less than 25".

Or we could leave it undocumented! The original intent was to just catch the exploratory queries that users reliably typed anyways, instead of steering them to this query pattern. But perhaps that ship has sailed...

Reply:
I just updated this PR at the exact same second you posted this comment :) Let me know what you think of the new copy @bkirwi . Was thinking it's better to point users to the efficient path since we have it, if they're looking to do data exploration anyways. But let me know if you'd rather we not mention it so explicitly

Reply:
Incredible timing! The updated copy seems right - slightly more restrictive than the feature actually is, but in ways that would take longer to explain and it's unlikely users care about 😅.

I'm still a bit on the fence about how much we want to highlight this, but I'd be happy going with what you have! Can always snip it out if it turns out users are getting confused.

Reply:
Sounds good, thanks for reviewing!